lib/utils/sort-locales.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Given a list of locales, this utility will sort them by `fallback` order, i.e. the locales without fallbacks first,
+ * then the locales having them as fallbacks, and then recursively.
+ */
+export default function sortLocales (locales) {
+  const localeByFallback = {}
+
+  locales.forEach(locale => {
+    if (locale.fallbackCode === null) {
+      locale.fallbackCode = undefined
+    }
+
+    if (!localeByFallback[locale.fallbackCode]) {
+      localeByFallback[locale.fallbackCode] = []
+    }
+
+    localeByFallback[locale.fallbackCode].push(locale)
+  })
+
+  return sortByFallbackKey(undefined, localeByFallback)
+}
+
+function sortByFallbackKey (key, localeByFallback) {
+  if (!localeByFallback[key]) {
+    return []
+  } else {
+    const resolved = localeByFallback[key]
```

Comment:
Does `resolved` hold sorted locales in the long run? If yes, can we name this `sortedLocales` or something else that's more explicit maybe?

Reply:
Sounds good.